### ISSUE TITLE
`invalid-return-type` false positive when function returns decorator that uses the paramspec from the outer function

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ty#3317]
### `main.py`

```py
from typing import Callable


def fn[**P]() -> Callable[P, None]:

    def decorator(*args: P.args, **kwargs: P.kwargs) -> None: ...

    return decorator
```
```
Return type does not match returned value: expected `[**P'return](**P'return) -> None`, found `def decorator(**P@decorator) -> None` (invalid-return-type) [Ln 8, Col 12]
```
i think it's treating the paramspec on the nested function as a separate paramspec or something. [for comparison, pyright correctly treats it as the same paramspec](https://pyright-play.net/?strict=true&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoDCAhgDYlEBGJApgLABQDDAJtcFMCgNoBUPACgF0AFAEooAWgB8hUuSrUu-ADRQAcmBTVBALib0ohqK3asAxriIxcwnkRBoAzjqj8AdPaeq%2BAawDuns6ubv6B4tLqmtQubrH6RlAg1DAAriAoxtQWIFa4QA)

### Version

ty 0.0.32 (4d1e1fc57 2026-04-20)

**Comment by dhruvmanila** [comment]
Thanks for the report. I think this is because the `P` in `fn` is an unbounded type variable while the `P` in `decorator` is bounded to the `decorator` function. This might be related to https://github.com/astral-sh/ty/issues/2723.

**Comment by sharifhsn** [comment]
Closed by https://github.com/astral-sh/ruff/pull/24909

Repro:

```py
from typing import Callable

def fn[**P]() -> Callable[P, None]:
    def decorator(*args: P.args, **kwargs: P.kwargs) -> None: ...
    return decorator
```

ty 0.0.35:

```
error[invalid-return-type]: Return type does not match returned value
 --> 3317.py:4:18
  |
3 | def fn[**P]() -> Callable[P, None]:
  |                  ----------------- Expected `[**P'return](**P'return) -> None` because of return type
4 |     def decorator(*args: P.args, **kwargs: P.kwargs) -> None: ...
5 |     return decorator
  |            ^^^^^^^^^ expected `[**P'return](**P'return) -> None`, found `def decorator(**P@decorator) -> None`

Found 1 diagnostic
```

ty 0.0.36:

```
All checks passed!
```
